Introduction/OptimizingDiffEqCode.ipynb

```diff
@@ -6,15 +6,17 @@
    "source": [
     "# Optimizing DiffEq Code\n",
     "\n",
-    "In this notebook we will walk through some of the main tools for optimizing your code in order to efficiently solve DifferentialEquations.jl. User-side optimizations are important because, for sufficiently difficult problems, most of the time will be spent inside of your `f` function (more efficient integrators are essentially strategies to reduce the required number of `f` calls to hit the error tolerance). The main ideas for optimizing your DiffEq calls, or any Julia function, is the following:\n",
+    "In this notebook we will walk through some of the main tools for optimizing your code in order to efficiently solve DifferentialEquations.jl. User-side optimizations are important because, for sufficiently difficult problems, most of the time will be spent inside of your `f` function, the function you are trying to solve. \"Efficient\" integrators are those that reduce the required number of `f` calls to hit the error tolerance. The main ideas for optimizing your DiffEq code, or any Julia function, are the following:\n",
     "\n",
-    "- Make it type-stable\n",
     "- Make it non-allocating\n",
-    "- Reduce redundent calculations\n",
-    "- Make use of BLAS calls\n",
     "- Use StaticArrays for small arrays\n",
+    "- Use broadcast fusion\n",
```

Comment:
Note that I added "broadcast fusion" and "optimize algorithm choice" to this list. I also tried to re-arrange the list to roughly reflect the order in which you discuss these points.